packages/client/src/types.ts

```diff
@@ -86,40 +85,36 @@ export type setTopicSessionDataFn = (
 /**
  * Topic types
  */
-type BaseTopic = {
+export enum TopicType {
+  PERMIT_CHECK, // A permit-check that is either an "OLO flow" or an "IMTR flow" check. IMTR checks are configured in `packages/imtr/src/config`.
+  PERMIT_FORM, // A permit-form - required when PERMIT_NEEDED - with the main focus on generating a PDF
+  REDIRECT, // A direct redirect to "OLO". We only use this to track the visitor count from amsterdam.nl
+}
+
+export type Topic = {
+  intro?: string;
   name: string;
   slug: string;
   text: {
     heading: string;
     locationIntro?: string;
   };
+  type: TopicType;
 };
 
-type IMTRTopic = {
-  hasIMTR: true;
-  intro?: string;
-  redirectToOlo?: false;
-} & BaseTopic;
-
-type OloTopic = {
-  hasIMTR: false;
-  intro?: string;
-  redirectToOlo?: false;
-} & BaseTopic;
-
-type RedirectToOloTopic = {
-  hasIMTR: false;
-  redirectToOlo: true;
-  intro?: undefined;
-} & BaseTopic;
-
-export type Topic = OloTopic | IMTRTopic | RedirectToOloTopic;
+// This is an imported topic from the Flo Legal api
+export type ApiTopic = {
+  name?: string;
+  path?: string;
+  permits: string[];
+  slug: string;
+};
 
 /**
  * Checker related types
  */
 export type Answer = {
-  formValue: string;
+  formValue?: string;
```

Comment:
Why did you make this optional?

Reply:
Because `formvalue` is only used inside Radio/checkbox questions (`value={formValue}`) and not in `input["text"]` (`value={userAnswer as string}`)